pkg/util/trace/impl/motrace/report_statement.go

```diff
@@ -36,6 +37,71 @@ var NilTxnID [16]byte
 var NilSesID [16]byte
 
 // StatementInfo implement export.IBuffer2SqlItem and export.CsvFields
+
+var _ IBuffer2SqlItem = (*StatementInfo)(nil)
+
+func StatementInfoNew(i Item, ctx context.Context) Item {
+	if s, ok := i.(*StatementInfo); ok {
+		// process the execplan
+		s.ExecPlan2Json(ctx)
+		// remove the plan
+		s.jsonByte = nil
+		s.ExecPlan = nil
+
+		// remove the TransacionID
+		s.TransactionID = NilTxnID
+		s.StatementTag = ""
+		s.StatementFingerprint = ""
+		s.Error = nil
+		s.RowsRead = 0
+		s.BytesScan = 0
+		s.ResultCount = 0
+		return s
+	}
+	return nil
+}
+func StatementInfoUpdate(existing, new Item) {
+
+	e := existing.(*StatementInfo)
+	n := new.(*StatementInfo)
+	// update the stats
+	e.Duration += n.Duration
+	e.Statement += n.Statement + "; "
+	// reponseAt is the last response time
+	e.ResponseAt = n.ResponseAt
+	// TODO: update the stats still json here
+	n.ExecPlan2Json(context.Background())
+	// e.statsJsonByte
```

Comment:
Why do we still need ExecPlan2Json?   We must remove this.      All we need is update statement stats, which should direct read from memory, no to/from json.

Reply:
Way too expensive.    All you need is to get 2 or 3 numbers out of this thing.

Reply:
Resolved. Do not need the json anymore